src/components/datatable/DataTable.js

```diff
@@ -1,384 +1,235 @@
                 {i18n.t(
                     'Data table is not supported when events are grouped on the server.'
                 )}
             </div>
         )
     }
+
+    return (
+        <>
+            <TableVirtuoso
+                context={tableContext}
+                components={TableComponents}
+                style={{ height, width: '100%' }}
+                data={rows}
+                fixedHeaderContent={() => (
+                    <DataTableRow>
+                        {headers.map(({ name, dataKey, type }, index) => (
+                            <DataTableColumnHeader
+                                className={styles.columnHeader}
+                                key={`${dataKey}-${index}`}
+                                onSortIconClick={sortData}
+                                sortDirection={
+                                    dataKey === sortField
+                                        ? sortDirection
+                                        : 'default'
+                                }
+                                sortIconTitle={i18n.t('Sort by {{column}}', {
+                                    column: name,
+                                })}
+                                onFilterIconClick={type && Function.prototype}
+                                showFilter={!!type && dataKey !== 'index'}
+                                name={dataKey}
+                                filter={
+                                    type && (
+                                        <FilterInput
+                                            type={type}
+                                            dataKey={dataKey}
+                                        />
+                                    )
+                                }
+                            >
+                                {name}
+                            </DataTableColumnHeader>
+                        ))}
+                    </DataTableRow>
+                )}
+                itemContent={(_, row) =>
+                    row.map(({ dataKey, value }) => (
+                        <DataTableCell
+                            key={`dtcell-${dataKey}`}
+                            className={cx(styles.dataCell, {
+                                [styles.lightText]:
+                                    dataKey === 'color' && isDarkColor(value),
+                            })}
+                            backgroundColor={dataKey === 'color' ? value : null}
+                            align="left"
+                        >
+                            {dataKey === 'color' ? value?.toLowerCase() : value}
+                        </DataTableCell>
+                    ))
+                }
+            />
+            {isLoading && (
+                <ComponentCover>
+                    <CenteredContent>
+                        <CircularLoader />
+                    </CenteredContent>
+                </ComponentCover>
+            )}
```

Comment:
The loading is meant to handle 2 specific situations where data is added to the table after the table is in place and has all the standard data. For example, the event layer typically has some custom fields that are fetched when opening the table, so those columns get added. And for earth engine layers, the aggregated data is generated when the table is opened, and this can take some time. I'll send you some videos of this.